R/plot_plate.R

```diff
@@ -0,0 +1,273 @@
+    geom_point(aes(fill = category), size = area_size * 19 - 1, shape = 21, color = "black", stroke = 0) +
+    scale_fill_manual(values = legend_mapping) +
+    theme_void() +
+    scale_x_continuous(limits = c(0, 1)) +
+    scale_y_continuous(limits = c(0, 1)) +
+    ggtitle(plot_title) +
+    theme(
+      aspect.ratio = aspect_ratio,
+      plot.title = element_text(hjust = 0.5, size = area_size * 20 + 5, vjust = -1),
+      legend.title = element_text(size = 0),
+      legend.text = element_text(size = 12, face = "bold"),
+      legend.background = element_rect(fill = "white", linewidth = 0),
+      legend.key = element_rect(fill = "white", color = "white")
+    )
+
+  if ((dev.size("px") / background_image_resolution)[1] < (dev.size("px") / background_image_resolution)[2]){
+    p <- p + theme(legend.position = "bottom")
+  }
+
+  if (plot_numbers) {
+    p <- p + geom_text(aes(label = numbers), size = area_size * 8 - 0.5, color = "black", vjust = 0.5, hjust = 0.5, fontface = "bold")
+  }
+
+  if (!plot_legend) {
+    p <- p + theme(legend.position = "none")
+  }
+
+  p
+}
+
+
+#' Plot counts in a 96-well plate
+#'
+#' It is a function to plot counts in a 96-well plate using a color scale.
+#' The function uses a color scale to map the counts to colors and plot them
+#' in a 96-well plate.
+#' The function uses the plot_plate function to plot the counts.
+#'
+#' @param plate The plate object with the counts data
+#' @param analyte_name The name of the analyte
+#' @param plot_counts Logical indicating if the counts should be plotted
+#' @param plot_legend Logical indicating if the legend should be plotted
+#'
+#' @return A ggplot object
+#'
+#' @examples
+#' plate_filepath <- system.file("extdata", "CovidOISExPONTENT_CO.csv", package = "PvSTATEM", mustWork = TRUE)
+#' layout_filepath <- system.file("extdata", "CovidOISExPONTENT_CO_layout.xlsx", package = "PvSTATEM", mustWork = TRUE)
+#' plate <- read_luminex_data(plate_filepath, layout_filepath)
+#' plot_counts(plate = plate, analyte_name = "OC43_NP_NA", plot_counts = T, plot_legend = F)
+#'
+#' @export
+plot_counts <- function(plate, analyte_name = NULL, plot_counts = FALSE, plot_legend = FALSE) {
+
+  if (is.null(plate)) {
+    stop("The plate object must be provided")
+  }
+
+  if (is.null(plate$data)) {
+    stop("The plate object must have data")
+  }
+
+  if (is.null(analyte_name)){
+    stop("The analyte_name must be provided")
+  }
+
+  counts <- plate$get_data(analyte_name, data_type = "Count")
+  location <- plate$sample_locations
+  counts <- create_vector_without_holes(counts, location)
+
+  if (length(counts) != 96) {
+    stop("The counts vector must have 96 elements")
+  }
+
+  # mapping function from counts to colors
+  map_to_color <- function(count) {
+    if (count < 0) {
+      return("white")
+    }
+
+    if (count < 50) {
+      return("#cc3232")
+    } else if (count >= 50 && count <= 70) {
+      return("#e5e50f")
+    } else {
+      return("#2dc937")
+    }
+  }
```

Comment:
I would use the values from the `color_map` vector and not retype the hex codes.